### ISSUE TITLE
list-devices json or csv output

### ISSUE THREAD

**Issue by zxfrank** [no-fee-ethereum-mining/nsfminer#90]
The output of --list-devices is a little hard to parse and a JSON or CSV alternative would be easier

**Comment by Jorge-Moreira** [comment]
@jean-m-cyr wouldn't this make more sense to be a feature of the API? It seems weird to obtain this kind of data from the console.

**Comment by zxfrank** [comment]
I may be wrong but since the miner isn't running at this stage, the api wouldn't be available...

Usage would be listing the devices to determine which one is cuda and which are cl before launching the workers.

**Comment by Jorge-Moreira** [comment]
Got it!
Would it be nice to launch the miner using a csv or json file?

**Comment by Jorge-Moreira** [comment]
Just in case @zxfrank , until the issue gets done, here's a way to get the devices for CUDA ( `-UL` ) or OpenCL ( `-GL` ):
![image](https://user-images.githubusercontent.com/24962230/108438151-8fc10180-7246-11eb-832e-a307c5ae3cb0.png)

**Comment by zxfrank** [comment]
I managed it through my node service like this:

```javascript
function listCards(cback){
    exec(__dirname + '/bin/ethminer --list-devices', function (err, stdout, stderr) {
        //  Id Pci Id       Type Name                          CUDA SM   Total Memory 
        // --- ------------ ---- ----------------------------- ---- ---  ------------ 
        //   0 0000:01:00.0 Gpu  GeForce GTX 1660 Ti           Yes  7.5       5.81 GB
        let lines = stdout.replace('Total Memory', 'Total_Memory').replace('Pci Id', 'Pci_Id').split('\n'), lengths = [];
        lines[0] = lines[0].replace(/\s+/g, ',').split(',');
        if (lines[0][0] == '') lines[0].shift();
        lines[1] = lines[1].replace(/\s+/g, ',').split(',');
        let fields = {};
        for (let i = 0; i < lines[0].length; i++) {
            if (!lines[1][i]) continue;
            fields[lines[0][i].toUpperCase()] = lines[1][i].length;
        }
        //console.log(fields);
        let _cards = [];
        for (let i = 2; i < lines.length; i++) {
            let _l = lines[i] + '', _c = {};
            if (_l == '') continue;
            for (let x of Object.keys(fields)) {
                _c[x] = _l.substr(0, fields[x]).trim();
                _l = _l.substr(fields[x] + 1)
            }
            _c.PCI_ID = _c.PCI_ID.toUpperCase();
            _cards.push(_c);

        }
        cback(_cards)
    });
}
```

gives me an array like:

```
[{
  ID: '0',
  PCI_ID: '0000:01:00.0',
  TYPE: 'Gpu',
  NAME: 'GeForce RTX 3070',
  CUDA: 'Yes',
  SM: '8.6',
  CL: 'Yes',
  TOTAL_MEMORY: '7.80 G'
}]
```

I don't like that. But it works...
